### ISSUE TITLE
github on tags: true AND prerelease: true

### ISSUE THREAD

**Issue by andykais** [travis-ci/travis-ci#9930]
I have this workflow for npm publishing:
create github release, trigger travis deploy provider npm

Github has an option to label pre-releases. I like this idea because it lets me do all my package versioning (including beta package versioning) from within github. That is why I would like the ability to distiguish prereleases from regular releases within travis. That way, I can tag a npm "beta" release like this: `npm publish --tag next`

I see two possible solutions:
- travis includes a prerelease conditional in the travis.yml config
- set an environment variable like `GITHUB_PRERELEASE=<bool>`

[github release events](https://developer.github.com/v3/repos/releases/#get-a-single-release)  include a `prerelease` boolean, assuming travis is using webhooks this information is already available.

**Comment by vladak** [comment]
This is I believe similar to #9935.

**Comment by BanzaiMan** [comment]
Setting up two deployments, each responding to different [conditions](https://docs.travis-ci.com/user/deployment/#conditional-releases-with-on) should work for you.

**Comment by vladak** [comment]
Yes, that's what we use currently however this is far from being optimal as
the deployment configs are nearly identical - they differ just in the
condition distinguishing between release amd pre-release. In programming
this would be called unnecessary code duplication.

Dne so 6. 10. 2018 1:13 uživatel Hiro Asari <notifications@github.com>
napsal:
